### PR TITLE
rely on translation of deprecated X-Header-Format to HTTP_X_HEADER_FORMAT

### PR DESCRIPTION
**Pull request by machisuji**
has to go into 4.3 too and will as soon as 4.2 is merged into 4.3

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://machisuji-fix-x-auth-scheme-legacy-support.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://machisuji-fix-x-auth-scheme-legacy-support.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: lib/open_project/authentication.rb [hunk 1]
```diff
@@ -83,7 +83,12 @@ module WWWAuthenticate
       module_function
 
       def pick_auth_scheme(supported_schemes, default_scheme, request_headers = {})
-        req_scheme = request_headers['X-Authentication-Scheme']
+        # Check two keys in case rack or some other middleware translates the
+        # deprecated 'X-Header-Format'. This will be obsolete with rails 4 (and its
+        # more recent rack version) where the deprecated headers will all be mapped to
+        # the HTTP_HEADER_FORMAT.
```
Comment (Azure7111) [inline lib/open_project/authentication.rb]: If this turns obsolete with Rails 4 I'd suggest to turn this comment into a TODO, that way it _might_ not be forgotten as easily.

**Comment by machisuji** [comment]
Found a better way. The mapping already happens with rack 1.4.6 apparently. It's just that both the deprecated _and_ the mapped keys exist in **env**.

**Comment by machisuji** [comment]
This will do.

**Comment by Azure7111** [comment]
Those spec failures seem legit, I think you might have to touch the tests.

**Comment by machisuji** [comment]
'tis green.

**Pull request merged**